src/tests/test_utils/unit_test_utils/mocks/mock_engine/mock_plugin.cpp

```diff
@@ -190,3 +190,8 @@ OPENVINO_PLUGIN_API void InjectPlugin(ov::IPlugin* target) {
     std::lock_guard<std::mutex> lock(targets_mutex);
     targets.push(std::make_shared<MockInternalPlugin>(target));
 }
+
+OPENVINO_PLUGIN_API void ClearTargets() {
+    decltype(targets) empty;
+    std::swap(targets, empty);
```

Comment:
Why is swap required?
```suggestion
    targets = {}
```
It should release all items from queue. Should lock be applied as `targets` is global object and when modified is protected by lock in `create_plugin_engine` function?

Reply:
done